core/new-gui/src/app/workspace/service/drag-drop/drag-drop.service.ts

```diff
@@ -80,6 +87,21 @@ export class DragDropService {
         const operator = this.workflowUtilService.getNewOperatorPredicate(value.operatorType);
         // add the operator
         this.workflowActionService.addOperator(operator, value.offset);
+
+        if (this.suggestionOperator !== undefined) {
+          if (this.isLeft !== undefined && this.isLeft && operator.inputPorts.length > 0) {
+            const linkCell = this.workflowUtilService.getNewOperatorLink(this.suggestionOperator, operator);
+            this.workflowActionService.addLink(linkCell);
+          } else if (this.isLeft !== undefined && !this.isLeft && operator.outputPorts.length > 0) {
+            const linkCell = this.workflowUtilService.getNewOperatorLink(operator, this.suggestionOperator);
+            this.workflowActionService.addLink(linkCell);
+          }
+          const operatorID = this.suggestionOperator.operatorID;
+          const status = true;
+          this.operatorSuggestionUnhighlightStream.next({status, operatorID});
+          this.suggestionOperator = undefined;
+          this.isLeft = undefined;
+        }
```

Comment:
What does this section do? Add some comments describe what kind of operations is being done right here.

Also, 

`          const operatorID = this.suggestionOperator.operatorID;
          const status = true;
          this.operatorSuggestionUnhighlightStream.next({status, operatorID});
`

can be shortened to
`this.operatorSuggestionUnhighlightStream.next({status: true, operatorID: this.suggestionOperator.operatorID});`

Since `getNewOperatorLink()` is only used in this function, just include the implementation in here instead of defining this new function (remove `getNewOperatorLink()`). Rather, define a new function inside this service that takes a source and target operator, and it will check which port the operator will connect to (Eg: output-0, output-1, input-0, input-1), create a new link ID, and call `addLink` to add the new link to our workflow.